src/types/text_imput_v3.zig

```diff
@@ -0,0 +1,114 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const TextInputV3Manager = extern struct {
+    global: *wl.Global,
+    text_inputs: wl.list.Head(TextInputV3Manager, "link"),
+
+    display_destroy: wl.Listener(*wl.Display),
+
+    events: extern struct {
+        text_input: wl.Signal(*wlr.TextInputV3),
+        destroy: wl.Signal(*wlr.TextInputV3Manager),
+    },
+
+    extern fn wlr_text_input_v3_create(display: *wl.Display) ?*wlr.TextInputV3Manager;
+    pub fn create(display: *wl.Display) !*wlr.TextInputV3Manager {
+        return wlr_text_input_v3_create(display) orelse error.OutOfMemory;
+    }
+};
+
+pub const TextInputV3 = extern struct {
+    // Not sure about this
+    pub const Features = extern struct {
+        pub const surrounding_text = 1 << 0;
+        pub const content_type = 1 << 1;
+        pub const cursor_rectangle = 1 << 2;
+    };
+
+    pub const State = extern struct {
+        pub const Surrounding = extern struct {
+            text: ?[*:0]u8,
+            cursor: u32,
+            anchor: u32,
+        };
+    };
+
+    pub const ContentType = extern struct {
+        hint: u32,
+        purpose: u32,
+    };
+
+    pub const CursorRectangle = extern struct {
+        x: i32,
+        y: i32,
+        width: i32,
+        height: i32,
+    };
+
+    text_change_cause: u32,
+    features: u32,
+
+    seat: *wlr.Seat,
+    ressource: *wl.Ressource,
+    focused_surface: *wlr.Surface,
+    pending: wlr.TextInputV3.State,
+    current: wlr.TextInputV3.State,
+    current_serial: u32,
+    pending_enabled: bool,
+    current_enabled: bool,
+    active_features: u32,
+
+    link: wl.list.Link,
+
+    surface_destroy: wl.Listener(*wlr.Surface),
+    seat_destroy: wl.Listener(*wlr.Seat),
+
+    events: extern struct {
+        enable: wl.Signal(*wlr.TextInputV3),
+        commit: wl.Signal(*wlr.TextInputV3),
+        disable: wl.Signal(*wlr.TextInputV3),
+        enable: wl.Signal(*wlr.TextInputV3),
+    },
+};
+
+extern fn wlr_text_input_v3_send_enter(text_input: *wlr.TextInputV3, surface: *wlr.Surface) void;
```

Comment:
This and the other `wlr_text_input_v3` functions should go inside the `TextInputV3` namespace.